### PR TITLE
benchmark: add Sysbench v4 file

### PR DESCRIPTION
**Pull request by huangxiuyan**
via: https://github.com/pingcap/docs/blob/master/benchmark/sysbench-v4.md PTAL @lilin90

### CODE REVIEW

File: benchmark/sysbench-v4.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+```
+create database sbtest;
+```
+
+调整 Sysbench 脚本创建索引的顺序。Sysbench 按照“建表->插入数据->创建索引”的顺序导入数据。该方式对于 TiDB 需要花费更多的导入时间。用户可以通过调整顺序来加速数据的导入。
+
+假设用户使用的 [Sysbench](https://github.com/akopytov/sysbench/tree/1.0.14) 版本。 我们可以通过以下两种方式来修改。
+
+1. 直接下载为 TiDB 修改好的 [oltp_common.lua](https://raw.githubusercontent.com/pingcap/tidb-bench/master/sysbench-patch/oltp_common.lua) 文件，覆盖 `/usr/share/sysbench/oltp_common.lua` 文件。 
+2. 将 `/usr/share/sysbench/oltp_common.lua` 的第 [235](https://github.com/akopytov/sysbench/blob/1.0.14/src/lua/oltp_common.lua#L235) 行到第[240](https://github.com/akopytov/sysbench/blob/1.0.14/src/lua/oltp_common.lua#L240) 移动到第198行以后。
```
Comment (lilin90) [inline benchmark/sysbench-v4.md]:
```suggestion
2. 将 `/usr/share/sysbench/oltp_common.lua` 的第 [235](https://github.com/akopytov/sysbench/blob/1.0.14/src/lua/oltp_common.lua#L235) 行到第 [240](https://github.com/akopytov/sysbench/blob/1.0.14/src/lua/oltp_common.lua#L240) 行移动到第 198 行以后。
```

File: benchmark/sysbench-v4.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+MySQL 客户端执行以下 SQL，创建数据库 `sbtest`：
+
+```
+create database sbtest;
+```
+
+调整 Sysbench 脚本创建索引的顺序。Sysbench 按照“建表->插入数据->创建索引”的顺序导入数据。该方式对于 TiDB 需要花费更多的导入时间。用户可以通过调整顺序来加速数据的导入。
+
+假设用户使用的 [Sysbench](https://github.com/akopytov/sysbench/tree/1.0.14) 版本。 我们可以通过以下两种方式来修改。
```
Comment (lilin90) [inline benchmark/sysbench-v4.md]:
```suggestion
假设用户使用的 [Sysbench](https://github.com/akopytov/sysbench/tree/1.0.14) 版本。我们可以通过以下两种方式来修改。
```

File: benchmark/sysbench-v4.md [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+block-cache-size = "24GB"
+[rocksdb.writecf]
+block-cache-size = "6GB"
+```
+
+更详细的 TiKV 参数调优请参考[ TiKV 性能参数调优](https://pingcap.com/docs/op-guide/tune-tikv/)。
+ 
+## 测试过程
+
+> **注意:** 此次测试并没有使用如 HAproxy 等负载均衡工具。在 TiDB 单一节点上进行 Sysbench 测试，并把结果相加。负载均衡工具和不同版本参数也会影响性能表现。
```
Comment (lilin90) [inline benchmark/sysbench-v4.md]:
```suggestion
> **注意**：此次测试并没有使用如 HAproxy 等负载均衡工具。在 TiDB 单一节点上进行 Sysbench 测试，并把结果相加。负载均衡工具和不同版本参数也会影响性能表现。
```

File: benchmark/sysbench-v4.md [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+| oltp_read_only | 3\*16 | 3873.53 | 61976.50 | 12.25 | 16.12 | 56.94 |
+| oltp_read_only | 3\*32 | 5066.88 | 81070.16 | 19.42 | 26.20 | 123.41 |
+| oltp_read_only | 3\*64 | 5466.36 | 87461.81 | 34.65 | 63.20 | 231.19 |
+| oltp_read_only | 3\*128 | 6684.16 | 106946.59 | 57.29 | 97.55 | 180.85 |
+ 
+![](../media/oltp_read_only.png)
+
+## 常见问题
+
+### 在高并发压力下，TiDB、TiKV 的配置都合理，为什么整体性能还是偏低
```
Comment (yikeke) [inline benchmark/sysbench-v4.md]:
```suggestion
### 在高并发压力下，TiDB、TiKV 的配置都合理，为什么整体性能还是偏低？
```

File: benchmark/sysbench-v4.md [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+
+这种情况可能与使用了 proxy 有关。可以尝试直接对单个 TiDB 加压，将结果加和与使用 proxy 的情况进行对比。
```
Comment (yikeke) [inline benchmark/sysbench-v4.md]:
```suggestion
这种情况可能与使用了 proxy 有关。可以尝试直接对单个 TiDB 加压，将求和后的结果与使用 proxy 的情况进行对比。
```

File: benchmark/sysbench-v4.md [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+
+TiKV 虽然整体 CPU 偏低，但部分模块的 CPU 可能已经达到了很高的利用率。
+
+TiKV 的其他模块，如 storage readpool、coprocessor 和 grpc 的最大并发度限制是可以通过 TiKV 的配置文件进行调整的。
+
+通过 Grafana 的 TiKV Thread CPU 监控面板可以观察到其实际使用率。如出现多线程模块瓶颈，可以通过增加该模块并发度进行调整。
+
+### 在高并发压力下，TiKV 也未达到 CPU 使用瓶颈，为什么 TiDB 的 CPU 利用率依然很低？
+
+在某些高端设备上，使用的是 NUMA 架构的 CPU，跨 CPU 访问远端内存将极大损耗降低性能。TiDB 默认将使用服务器所有 CPU，goroutine 的调度不可避免的会出现跨 CPU 内存访问。
```
Comment (yikeke) [inline benchmark/sysbench-v4.md]:
```suggestion
在某些高端设备上，使用的是 NUMA 架构的 CPU，跨 CPU 访问远端内存将极大降低性能。TiDB 默认将使用服务器所有 CPU，goroutine 的调度不可避免的会出现跨 CPU 内存访问。
```

File: benchmark/sysbench-v4.md [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+
+因此，建议在 NUMA 架构服务器上，部署 *n* 个 TiDB（*n* = NUMA CPU 的个数），同时将 TiDB 的 `max-procs` 设置为一个 NUMA CPU 的核数。
```
Comment (yikeke) [inline benchmark/sysbench-v4.md]:
```suggestion
因此，建议在 NUMA 架构服务器上，部署 *n* 个 TiDB（*n* = NUMA CPU 的个数），同时将 TiDB 的 `max-procs` 变量的值设置为与 NUMA CPU 的核数相同。
```

File: benchmark/sysbench-v4.md [hunk 8]
```diff
@@ -0,0 +1,246 @@
+---
+title: TiDB 3.0 Beta 版本 Sysbench 测试结果 - NVMe SSD
```
Comment (yikeke) [inline benchmark/sysbench-v4.md]:
```suggestion
title: TiDB 3.0 Beta 版本在 NVMe SSD 上的 Sysbench 测试结果
```

File: benchmark/sysbench-v4.md [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+category: benchmark
+---
+
+#  TiDB 3.0 Beta 版本 Sysbench 测试结果 - NVMe SSD
```
Comment (yikeke) [inline benchmark/sysbench-v4.md]:
```suggestion
#  TiDB 3.0 Beta 版本在 NVMe SSD 上的 Sysbench 测试结果
```

File: benchmark/sysbench-v4.md [hunk 10]
```diff
@@ -0,0 +1,246 @@
...
+```
+
+调整 Sysbench 脚本创建索引的顺序。Sysbench 按照“建表->插入数据->创建索引”的顺序导入数据。该方式对于 TiDB 需要花费更多的导入时间。用户可以通过调整顺序来加速数据的导入。
+
+假设用户使用的 [Sysbench](https://github.com/akopytov/sysbench/tree/1.0.14) 版本。我们可以通过以下两种方式来修改。
+
+1. 直接下载为 TiDB 修改好的 [oltp_common.lua](https://raw.githubusercontent.com/pingcap/tidb-bench/master/sysbench-patch/oltp_common.lua) 文件，覆盖 `/usr/share/sysbench/oltp_common.lua` 文件。 
+2. 将 `/usr/share/sysbench/oltp_common.lua` 的第 [235](https://github.com/akopytov/sysbench/blob/1.0.14/src/lua/oltp_common.lua#L235) 行到第[240](https://github.com/akopytov/sysbench/blob/1.0.14/src/lua/oltp_common.lua#L240) 移动到第 198 行以后。
+
+> **注意:** 此操作为可选操作，仅节约了数据导入时间。
```
Comment (CaitinChen) [inline benchmark/sysbench-v4.md]:
```suggestion
> **注意**：此操作为可选操作，仅节约了数据导入时间。
```

Comment (CaitinChen) [reply benchmark/sysbench-v4.md]: Your semicolon is an English character.

File: benchmark/sysbench-v4.md [hunk 11]
```diff
@@ -0,0 +1,246 @@
...
+```
+
+调整 Sysbench 脚本创建索引的顺序。Sysbench 按照“建表->插入数据->创建索引”的顺序导入数据。该方式对于 TiDB 需要花费更多的导入时间。用户可以通过调整顺序来加速数据的导入。
+
+假设用户使用的 [Sysbench](https://github.com/akopytov/sysbench/tree/1.0.14) 版本。我们可以通过以下两种方式来修改。
+
+1. 直接下载为 TiDB 修改好的 [oltp_common.lua](https://raw.githubusercontent.com/pingcap/tidb-bench/master/sysbench-patch/oltp_common.lua) 文件，覆盖 `/usr/share/sysbench/oltp_common.lua` 文件。 
+2. 将 `/usr/share/sysbench/oltp_common.lua` 的第 [235](https://github.com/akopytov/sysbench/blob/1.0.14/src/lua/oltp_common.lua#L235) 行到第[240](https://github.com/akopytov/sysbench/blob/1.0.14/src/lua/oltp_common.lua#L240) 移动到第 198 行以后。
+
+> **注意**： 此操作为可选操作，仅节约了数据导入时间。
```
Comment (CaitinChen) [inline benchmark/sysbench-v4.md]:
```suggestion
> **注意**：此操作为可选操作，仅节约了数据导入时间。
```

**Comment by CaitinChen** [comment]
@xiekeyi98 PTAL

**Comment by CaitinChen** [comment]
@crazycs520 PTAL Thanks.

**Comment by CaitinChen** [comment]
@jackysp PTAL

**Pull request merged**